src/net/mod.rs

```diff
@@ -62,6 +79,45 @@ mod conn {
     }
 }
 
+
+#[cfg(target_arch = "wasm32")]
+mod wasm_connector {
+    use super::*;
+    use std::io::Error as IoError;
+    use async_trait::async_trait;
+    use ws_stream_wasm::WsMeta;
+    #[derive(Clone, Default)]
+    pub struct DefaultDomainWebsocketConnector {
+    }
+    impl DefaultDomainWebsocketConnector {
+        pub fn new() -> Self {
+            Self {}
+        }
+    }
+    #[async_trait(?Send)]
+    impl TcpDomainConnector for DefaultDomainWebsocketConnector {
+        async fn connect(
+            &self,
+            addr: &str,
+        ) -> Result<(BoxWriteConnection, BoxReadConnection, ConnectionFd), IoError> {
+            let (mut _ws, wsstream) = WsMeta::connect(addr, None).await.unwrap();
+            // wsstream implements both AsyncRead and AsyncWrite but there's not a good way to
+            // split them.
+            let _wsstream = wsstream.into_io();
```

Comment:
@sehz this `IoStream` implements both `AsyncRead` and `AsyncWrite` but I'm struggling to split them.

Reply:
And there is https://docs.rs/futures/0.3.15/futures/lock/struct.BiLock.html